src/sdk/accountspace_fs.js

```diff
@@ -776,37 +838,22 @@ class AccountSpaceFS {
         }
     }
 
-    async _create_access_key_symlink(requested_account_name, access_key_id) {
-        const account_config_relative_path = get_config_file_path(acounts_dir_relative_path, requested_account_name);
-        const new_access_key_symlink_config_path = get_symlink_config_file_path(this.access_keys_dir, access_key_id);
-        await nb_native().fs.symlink(this.fs_context, account_config_relative_path, new_access_key_symlink_config_path);
+    async _symlink_to_account(account_id, symlink_dir, symlink_filename, root_name) {
+        const account_config_relative_path = get_config_file_path(
+            //if root_name, then we need account_name->account, which is two dirs above.
+            //otherwise, only one dir above
+            root_name ? acounts_dir_relative_path_double : acounts_dir_relative_path,
+            account_id);
+        const symlink_path = get_symlink_config_file_path(symlink_dir, symlink_filename, root_name);
+        await nb_native().fs.symlink(this.fs_context, account_config_relative_path, symlink_path);
     }
 
-    async _update_account_config_new_username(action, params, requested_account) {
-        await this._check_username_already_exists(action, params.new_username);
-        // prepare
-        requested_account.name = params.new_username;
-        requested_account.email = params.new_username; // internally saved
-        const access_key_ids = [];
-        for (const access_keys of requested_account.access_keys) {
-            access_key_ids.push(access_keys.access_key);
-        }
-        // handle account config creation
-        const requested_account_encrypted = await nc_mkm.encrypt_access_keys(requested_account);
-        const account_string = JSON.stringify(requested_account_encrypted);
-        nsfs_schema_utils.validate_account_schema(JSON.parse(account_string));
-        const new_username_account_config_path = this._get_account_config_path(params.new_username);
-        await native_fs_utils.create_config_file(this.fs_context, this.accounts_dir,
-            new_username_account_config_path, account_string);
-        // handle access keys (unlink and then create the new symbolic link)
-        for (const access_key_id of access_key_ids) {
-            const requested_account_path = get_symlink_config_file_path(this.access_keys_dir, access_key_id);
-            await nb_native().fs.unlink(this.fs_context, requested_account_path);
-            this._create_access_key_symlink(params.new_username, access_key_id);
-        }
-        // handle account config deletion
-        const account_config_path = this._get_account_config_path(params.username);
-        await native_fs_utils.delete_config_file(this.fs_context, this.accounts_dir, account_config_path);
+    async _update_account_config_new_username(action, user_id, old_username, new_username, requesting_account) {
```

Comment:
I'm not sure it's related I saw an issue on renaming a root account:
1. create an IAM roots account manager in the CLI
2. create root accounts in the IAM API
3. rename the root account with the IAM API

> Jul-31 17:09:42.837 [nsfs/47471] [ERROR] core.sdk.accountspace_fs:: AccountSpaceFS.update_user error [Error: No such file or directory] { code: 'ENOENT', context: 'Unlink _path=/etc/noobaa.conf.d/root_accounts/manager/my-root.symlink ' }

Reply:
The rename was taking the root account manager name instead of the requested account.
I pushed a fix now.